text/src/autogluon/text/automm/data/infer_types.py

```diff
@@ -0,0 +1,301 @@
+    it will be treated as a categorical column.
+
+    Parameters
+    ----------
+    data
+        The column data
+    valid_data
+        Additional validation data
+    threshold
+        The threshold for detecting categorical column
+    ratio
+        The ratio detecting categorical column
+    oov_ratio_threshold
+        The out-of-vocabulary ratio between training and validation.
+        This is used to determine if the column is a categorical column.
+        Usually, a categorical column can tolerate a small OOV ratio
+    is_label
+        Whether the column is a label column.
+
+    Returns
+    -------
+    is_categorical
+        Whether the column is a categorical column
+    """
+    if data.dtype.name == 'category':
+        return True
+    else:
+        if threshold is None:
+            if is_label:
+                threshold = 100
+                oov_ratio_threshold = 0
+                ratio = 0.1
+            else:
+                threshold = 20
+                oov_ratio_threshold = 0
+                ratio = 0.1
+        threshold = min(int(len(data) * ratio), threshold)
+        data_value_counts = data.value_counts(dropna=False)
+        key_set = set(data_value_counts.keys())
+        if len(data_value_counts) < threshold:
+            valid_value_counts = valid_data.value_counts(dropna=False)
+            total_valid_num = len(valid_data)
+            oov_num = 0
+            for k, v in zip(valid_value_counts.keys(), valid_value_counts.values):
+                if k not in key_set:
+                    oov_num += v
+            if is_label and oov_num != 0:
+                return False
+            if oov_num / total_valid_num > oov_ratio_threshold:
+                return False
+            return True
+        return False
+
+
+def is_numerical_column(data: pd.Series,
+                        valid_data: Optional[pd.Series] = None,
+                        ) -> bool:
+    """Try to identify if a column is a numerical column.
+
+    We adopted a very simple rule to verify if the column is a numerical column.
+
+    Parameters
+    ----------
+    data
+        The training data series
+    valid_data
+        The validation data series
+
+    Returns
+    -------
+    is_numerical
+        Whether the column is a numerical column
+    """
+    try:
+        numerical_data = pd.to_numeric(data)
+        if valid_data is not None:
+            numerical_valid_data = pd.to_numeric(valid_data)
+        return True
+    except:
+        return False
+
+
+def is_imagepath_column(data: pd.Series,
+                        ) -> bool:
+    """Try to identify if a column is a image column.
+    We adopted a very simple rule to verify if the column is a image column.
+    Parameters
+    ----------
+    data
+        The training data series
+    Returns
+    -------
+    is_image
+        Whether the column is a image column
+    """
+    try:
+        if (data.iloc[0].startswith('/') or data.iloc[0].startswith('.')):
```

Comment:
Consider to revise the logic. This essentially detects if the data contain paths but these paths may not be images. We may sample a few entries from the column, use PIL to open them and check the success ratio.

Reply:
Done.